frontend/src/components/MainComponent/MainComponent.tsx

```diff
@@ -184,6 +184,9 @@ function MainComponent({
 			return defence;
 		});
 		setDefencesToShow(newDefences);
+		// add info message to chat
+		const displayedDefenceId = defenceId.replace(/_/g, ' ').toLowerCase();
+		addInfoMessage(`${displayedDefenceId} defence reset`);
```

Comment:
You have these two lines duplicated now; may as well pull out into a separate function, which takes param 'reset' or 'configured' for the message to add.